### PR TITLE
Add optional way to extend the build

### PR DESCRIPTION
**Pull request by cmcfarlen**
Add ext dir and option to extend the build to add plugins or cripts bundles

### CODE REVIEW

**Review by zwoop — APPROVED** [review]

File: ext/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+file(
+  GLOB EXT_SUBDIRS
+  LIST_DIRECTORIES true
+  *
+)
+
+foreach(SDIR ${EXT_SUBDIRS})
+  if(IS_DIRECTORY ${SDIR})
+    add_subdirectory(${SDIR})
```
Comment (JosiahWI) [inline ext/CMakeLists.txt:34]: What if the directory doesn't contain a CMakeLists.txt?

Comment (cmcfarlen) [reply ext/CMakeLists.txt:34]: I can check for that!  I guess they can be ignored, but maybe I'll print a warning

**Review by JosiahWI — APPROVED** [review]
Thanks for remembering to document in the README, looks good!

**Comment by JosiahWI** [comment]
The AuTests prefetch_simple and prefetch_overflow failed with each one missing the last GET request.

**Comment by JosiahWI** [comment]
[approve ci autest]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**